File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
         
        int Max=INT_MIN;
        int Min=INT_MAX;
        for(int i=0;i<prices.size();i++)
        {
           
            Min=min(Min,prices[i]);
            Max=max(Max,prices[i]-Min);  
        }
        return Max;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,r,profit=0,1,0
        while r <len(prices):
            if prices[l] < prices[r]:
                profit=max(prices[r] -prices[l],profit)
            else:
                l=r
            r+=1
        return profit
                            
                       

        